src/main/java/io/stargate/sgv2/jsonapi/api/model/command/deserializers/FilterClauseDeserializer.java

```diff
@@ -53,9 +65,89 @@ public FilterClause deserialize(
                 entry.getKey(), jsonNodeValue(entry.getKey(), entry.getValue())));
       }
     }
+
+    validate(expressionList);
     return new FilterClause(expressionList);
   }
 
+  private void validate(List<ComparisonExpression> expressionList) {
+    for (ComparisonExpression expression : expressionList) {
+      expression.filterOperations().forEach(operation -> validate(expression.path(), operation));
+    }
+  }
+
+  private void validate(String path, FilterOperation<?> filterOperation) {
+    if (filterOperation.operator() instanceof ValueComparisonOperator valueComparisonOperator) {
+      switch (valueComparisonOperator) {
+        case IN -> {
+          if (!path.equals(DocumentConstants.Fields.DOC_ID)) {
+            throw new JsonApiException(
+                ErrorCode.INVALID_FILTER_EXPRESSION, "Can use $in operator only on _id field");
+          }
+
+          if (filterOperation.operand().value() instanceof List<?> list) {
+            if (list.isEmpty()) {
+              throw new JsonApiException(
+                  ErrorCode.INVALID_FILTER_EXPRESSION, "$in operator must have at least one value");
+            }
+            if (list.size() > documentConfig.defaultPageSize()) {
+              throw new JsonApiException(
+                  ErrorCode.INVALID_FILTER_EXPRESSION,
+                  "$in operator must have at most " + documentConfig.defaultPageSize() + " values");
```

Comment:
20 is ok for now, but we will likely have to increase this before GA. It isn't uncommon for Mongoose `populate()` to send 50-100 entries in a `$in` in production.

Reply:
I am not Aaron but I suspect it makes sense to use separate limit -- but document that there is a co-dependency.